packages/graphql/src/translate/create-delete-and-params.ts

```diff
@@ -273,30 +316,105 @@ function createDeleteAndParams({
                             fromTypename,
                             toTypename,
                         });
-                        const reduceStr = `REDUCE(m=${META_CYPHER_VARIABLE}, n IN ${nodeToDelete} | m + ${metaObjectStr} + ${eventWithMetaStr}) AS ${META_CYPHER_VARIABLE}`;
-                        res.strs.push(
-                            `WITH ${[...filterMetaVariable(withVars), nodeToDelete].join(", ")}, ${reduceStr}`
+
+                        const CB_deleteHelperNode = new Cypher.Node();
+                        const CB_deleteMetaVar = new Cypher.NamedVariable("delete_meta");
+                        const CB_innerSubquery = new Cypher.Call(
+                            new Cypher.Unwind([CB_nodeToDelete, CB_deleteHelperNode])
+                                .with(...CB_withVars, CB_relationship, CB_deleteHelperNode, [
+                                    new Cypher.RawCypher(`[] + ${metaObjectStr} + ${eventWithMetaStr}`),
+                                    CB_metaVar,
+                                ])
+                                .detachDelete(CB_deleteHelperNode)
+                                .return([Cypher.collect(CB_metaVar), CB_deleteMetaVar])
+                        ).innerWith(CB_relationship, CB_nodeToDelete, ...CB_withVars);
+
+                        const CB_reduceAccVar = new Cypher.Variable();
+                        const CB_reduceIteratorVar = new Cypher.Variable();
+                        CB_subquery.concat(
+                            new Cypher.With(...CB_withVars, CB_metaVar, CB_relationship, [
+                                Cypher.collect(CB_node1), //TODO: DISTINCT
+                                CB_nodeToDelete,
+                            ]),
+                            CB_innerSubquery,
+                            new Cypher.With(CB_deleteMetaVar, CB_metaVar),
+                            new Cypher.Return([
+                                Cypher.reduce(
```

Comment:
Although not changed in this PR, it seems as though `Cypher.reduce` is producing upper case `REDUCE()` - this is a function so should be `reduce()`. We should change either here or in a separate PR!